File: find_min_max/Min_max.java
package find_min_max;

// { Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class pair  
{  
    private long first, second;  

    public pair(long first, long second)  
    {  
        this.first = first;  
        this.second = second;  
    }  

    public long getFirst(){
        return this.first;
    }

    public long getSecond(){
        return this.second;
    }
}

class Min_max {
	public static void main(String[] args) throws IOException
	{
        
	        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        // int t =
        //     Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        // while(t-->0)
        // {
            long n = Long.parseLong(br.readLine().trim());
            long a[] = new long[(int)(n)];
            // long getAnswer[] = new long[(int)(n)];
            String inputLine[] = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                a[i] = Long.parseLong(inputLine[i]);
            }
            
            Compute obj = new Compute();
            pair product = obj.getMinMax(a, n); 
            System.out.println("min = "+ product.getFirst() +" :: "+ "max = " + product.getSecond());
            
        // }
	}
}

// } Driver Code Ends

//User function Template for Java

/*
 class pair  
{  
    long first, second;  
    public pair(long first, long second)  
    {  
        this.first = first; min  
        this.second = second;  max
    }  
} */

class Compute 
{
    static pair getMinMax(long a[], long n)  
    {
        //Write your code here
        // long min = a[0];
        // long max = a[0];

        // for (int i = 0; i < n; i++) {
        //     if(a[i] < min)
        //         min = a[i];
        //     if(a[i] > max)
        //         max = a[i];
        // }
        // return new pair(min, max);

        // slower than above method but short
        TreeSet<Long> ts = new TreeSet<>();
        for (Long temp : a) {
            ts.add(temp);
        }
        return new pair(ts.first(), ts.last());
    }
}

File: Beginner's DSA sheet/find_min_max/min_max.py
#User function Template for python3

def getMinMax(a, n):
    max = a[0]
    min = a[0]
    # for temp in a:
    #     if temp > max:
    #         max = temp
    #     if temp < min:
    #         min = temp
    
    for i in range(n):
        if a[i] > max:
            max = a[i]
        if a[i] < min:
            min = a[i]
            
    return min, max
    # return min(a), max(a)
    
    
    enu
#{ 
#  Driver Code Starts
#Initial Template for Python 3

def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        
        product = getMinMax(a, n)
        print(product[0], end=" ")
        print(product[1])

        T -= 1


if __name__ == "__main__":
    main()



# } Driver Code Ends